syft/execution/role.py

```diff
@@ -69,15 +68,34 @@ def register_inputs(self, args_):
         """ Takes input arguments for this role and generate placeholders.
         """
         # TODO Should we be able to rebuild?
-        self.input_placeholder_ids = tuple(self._build_placeholders(arg).value for arg in args_)
+        def register_nested_inputs(obj) -> None:
+            if isinstance(obj, (list, tuple)):
+                _ = [register_nested_inputs(elem) for elem in obj]
+            elif isinstance(obj, dict):
+                _ = [register_nested_inputs(v) for _, v in sorted(obj.items())]
+            elif isinstance(obj, FrameworkTensor):
```

Comment:
This kind of nested collection traversal is a common pattern between this method and `register_nested_outputs`, and we use a similar pattern elsewhere in the code base. I wonder if we can extract a common traversal utility that accepts a function or a `type->function` dictionary to specify what to do with the leaf elements?

Reply:
Made one for Role, I could make it generic and usable in all classes, but might be awkard to use/to understand, I believe the one for Role is enough.